src/kernel/src/fs/uio.rs

```diff
@@ -34,6 +34,18 @@ impl<'a> IoVec<'a> {
             _phantom: std::marker::PhantomData,
         }
     }
+
+    pub fn from_slice(slice: &mut [u8]) -> Self {
```

Comment:
Not sure if Rust will bind the lifetime of `slice` as `'a`. If you also don't know better to change this to:

```rust
pub fn from_slice(slice: &'a mut [u8]) -> Self {}
```